testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jpa/deploymentsubsystemoperations/JPADeploymentSubsystemOperationsTestCase.java

```diff
@@ -0,0 +1,152 @@
+ */
+
+package org.jboss.as.test.integration.jpa.deploymentsubsystemoperations;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.PathElement;
+import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.as.controller.operations.common.Util;
+import org.jboss.as.test.shared.TestSuiteEnvironment;
+import org.jboss.dmr.ModelNode;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.IOException;
+
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.ADD;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.DEPLOYMENT;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.FAILED;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.FAILURE_DESCRIPTION;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OUTCOME;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.REMOVE;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.RESULT;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SUBSYSTEM;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.WRITE_ATTRIBUTE_OPERATION;
+
+/**
+ * Regression tests for WFLY-11173
+ * Tests proper operations behavior in /deployment/subsystem=jpa case
+ *
+ * @author <a href="mailto:pberan@redhat.com">Petr Beran
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+public class JPADeploymentSubsystemOperationsTestCase {
+
+    private static ModelControllerClient controllerClient = TestSuiteEnvironment.getModelControllerClient();
+
+    private static final PathAddress DEPLOYMENT_PATH = PathAddress.pathAddress(PathElement.pathElement(DEPLOYMENT, JPADeploymentSubsystemOperationsTestCase.class.getSimpleName() + ".war"));
+    private static final PathAddress SUBSYSTEM_PATH = PathAddress.pathAddress(PathElement.pathElement(SUBSYSTEM, "jpa"));
+
+    // Hardcoded to avoid adding the JPA module as a dependency
+    // Same goes for the "jpa" in SUBSYSTEM_PATH
+    private static final String DEFAULT_DATASOURCE = "default-datasource";
+    private static final String DEFAULT_EXTENDED_PERSISTENCE_INHERITANCE = "default-extended-persistence-inheritance";
+
+    @Deployment
+    public static Archive<?> deploy() {
+        WebArchive war = ShrinkWrap.create(WebArchive.class, JPADeploymentSubsystemOperationsTestCase.class.getSimpleName() + ".war");
+        war.addAsResource(JPADeploymentSubsystemOperationsTestCase.class.getPackage(), "persistence.xml", "META-INF/persistence.xml");
+        return war;
```

Comment:
This deployment is not sufficient to test this issue.
The JPA subsystem does not create /subsystem=jpa deployment resources unless it contains a PersistenceUnit.

Reply:
Is it really not sufficient? If the deployment subsystem resources are not present the tests fail as the deployment subsystem read-resource operation returns `undefined`. 

As long as there's `persistence-unit` in the persistence.xml, it gets picked by the [PersistenceUnitParseProcessor](https://github.com/wildfly/wildfly/blob/main/jpa/subsystem/src/main/java/org/jboss/as/jpa/processor/PersistenceUnitParseProcessor.java#L418) and is marked as a JPA Deployment and thus is handled by the [PersistenceUnitServiceHandler](https://github.com/wildfly/wildfly/blob/main/jpa/subsystem/src/main/java/org/jboss/as/jpa/processor/PersistenceUnitServiceHandler.java#L171)

Reply:
@petrberan You are indeed correct! :)
When I tested your branch locally, I had the persistence.xml mistakenly located within /META-INF/persistence.xml, instead of /WEB-INF/classes/META-INF/persistence.xml - which is the only valid location for a war.

I missed the logic with testReadResourceOperation(), which is the only "non-negative" test case - which, as you pointed out, would not succeed I my statement above had any validity...